File: examples/sample.py
"""
Sample Python entities
Use comment annotations: # @pk, # @unique, # @notNull, # @default, # @ref
"""

from dataclasses import dataclass
from datetime import datetime
from typing import Optional

@dataclass
class User:
    id: int  # @pk
    username: str  # @unique @notNull
    email: str  # @unique @notNull
    password_hash: str  # @notNull
    created_at: datetime
    updated_at: Optional[datetime] = None

@dataclass
class Post:
    id: int  # @pk
    title: str  # @notNull
    body: Optional[str] = None
    author_id: int  # @notNull @ref User.id many-to-one
    status: str = 'draft'  # @default draft
    published_at: Optional[datetime] = None
    created_at: datetime = None

@dataclass
class Comment:
    id: int  # @pk
    post_id: int  # @notNull @ref Post.id many-to-one
    user_id: int  # @notNull @ref User.id many-to-one
    content: str  # @notNull
    created_at: datetime = None

File: examples/sample.java
/**
 * Sample Java entities
 * Use JavaDoc annotations: @pk, @unique, @notNull, @default, @ref
 * Or JPA annotations: @Id, @Column, @ManyToOne, etc.
 */

import javax.persistence.*;
import java.time.LocalDateTime;

public class User {
    /**
     * @pk
     */
    @Id
    private Integer id;

    /**
     * @unique
     * @notNull
     */
    @Column(unique = true, nullable = false)
    private String username;

    /**
     * @unique
     * @notNull
     */
    private String email;

    /**
     * @notNull
     */
    private String passwordHash;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

public class Post {
    /**
     * @pk
     */
    @Id
    private Integer id;

    /**
     * @notNull
     */
    private String title;

    private String body;

    /**
     * @notNull
     * @ref User.id many-to-one
     */
    @ManyToOne
    private Integer authorId;

    /**
     * @default 'draft'
     */
    private String status; // draft, published, archived

    private LocalDateTime publishedAt;
    private LocalDateTime createdAt;
}

public class Comment {
    /**
     * @pk
     */
    @Id
    private Integer id;

    /**
     * @notNull
     * @ref Post.id many-to-one
     */
    @ManyToOne
    private Integer postId;

    /**
     * @notNull
     * @ref User.id many-to-one
     */
    @ManyToOne
    private Integer userId;

    /**
     * @notNull
     */
    private String content;

    private LocalDateTime createdAt;
}
